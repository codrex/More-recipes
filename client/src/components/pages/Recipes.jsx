import React from 'react';
import { connect } from 'react-redux';
import PropTypes from 'prop-types';
import { getAllRecipes, findRecipes } from '../../actions/recipeActions';
import RecipesDisplay from './RecipesDisplay';

/**
 * Recipes component
 */
class Recipes extends React.Component {
  /**
   * @return {undefined}
   * @param {object} nextProps
   */
  componentWillReceiveProps(nextProps) {
    const {
      searchTerm,
      search
    } = this.props;
    if (nextProps.searchTerm !== searchTerm) {
      return search(nextProps.searchTerm, 1);
    }
  }

  /**
   * @param {number} page
   * @return {undefined}
   */
  fetchRecipes = (page) => {
    const {
      searchTerm,
      search
    } = this.props;
    if (searchTerm) {
      return search(searchTerm, page);
    }
    this.props.getAllRecipes(page);
  }
  /**
   * @return {React} component
   */
  render() {
    return (
      <RecipesDisplay
        title="recipes"
        getRecipes={this.fetchRecipes}
        recipes={this.props.recipes}
        {...this.props}
      />
    );
  }
}

Recipes.defaultProps = {
  searchTerm: '',
};

Recipes.propTypes = {
  getAllRecipes: PropTypes.func.isRequired,
  recipes: PropTypes.arrayOf(PropTypes.shape).isRequired,
  searchTerm: PropTypes.string,
  search: PropTypes.func.isRequired,
};

const mapStateToProps = state => ({
  recipes: state.recipes,
});

export { Recipes as PureRecipes };
export default connect(mapStateToProps, {
  getAllRecipes,
  search: findRecipes
})(Recipes);
