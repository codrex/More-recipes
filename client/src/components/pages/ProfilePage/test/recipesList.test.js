import React from 'react';
import Recipes from '../recipesList/recipesList';
import toJson from 'enzyme-to-json';

const props = {
  recipes: [],
  onEditIconCliked: jest.fn(),
  handleClick: jest.fn(),
  onDeleteIconClicked: jest.fn(),
  onFavIconClicked: jest.fn(),
  type: 'createdRecipes'
};

describe('Recipes list component ', () => {
  test('to match empty snapshot ', () => {
    const wrapper = shallow(<Recipes {...props} />);
    const tree = toJson(wrapper);
    expect(tree).toMatchSnapshot();
    expect(tree).toBeInstanceOf(Object);
  });
  test('render as expected when created recipes nav link is clicked ', () => {
    const wrapper = mount(< Recipes {...{ ...props, recipes: [{}] }} />);
    let tree = toJson(wrapper);
    expect(tree).toMatchSnapshot();
    wrapper.find('ListItem').simulate('click');
    tree = toJson(wrapper);
    expect(tree).toMatchSnapshot();
  });
  test('render as expected when type equals favRecipes ', () => {
    const wrapper = mount(< Recipes {...{ ...props, recipes: [{}], type: 'favRecipes' }} />);
    let tree = toJson(wrapper);
    expect(tree).toMatchSnapshot();
    wrapper.find('ListItem').simulate('click');
    tree = toJson(wrapper);
    expect(tree).toMatchSnapshot();
  });
  test('render as expected when delete and edit icon are clicked ', () => {
    const wrapper = mount(< Recipes {...{ ...props, recipes: [{}] }} />);
    let tree = toJson(wrapper);
    expect(tree).toMatchSnapshot();
    wrapper.find('ListItem').find('Icon').at(0)
    .simulate('click');
    wrapper.find('ListItem').find('Icon').at(1)
    .simulate('click');
    tree = toJson(wrapper);
    expect(tree).toMatchSnapshot();
  });
});

