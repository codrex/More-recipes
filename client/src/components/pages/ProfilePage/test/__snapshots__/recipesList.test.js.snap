// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Recipes list component  render as expected when created recipes nav link is clicked  1`] = `
<Recipes
  handleClick={[Function]}
  onDeleteIconClicked={[Function]}
  onEditIconCliked={[Function]}
  onFavIconClicked={[Function]}
  recipes={
    Array [
      Object {},
    ]
  }
  type="createdRecipes"
>
  <List
    className="recipe-list"
    style={Object {}}
  >
    <div
      className="col-12 recipe-list"
      style={Object {}}
    >
      <h4
        className="display-4 bold header"
      >
        my recipes
      </h4>
      <ListItem
        className="recipe-list-item"
        handleClick={[Function]}
        key="0"
      >
        <div
          className="lead list-item recipe-list-item"
        >
          <div>
            <Icon
              handleClick={[Function]}
              iconClass="fa fa-pencil edit-icon"
              pointer={true}
            >
              <div
                className="icon-wrapper pointer"
                onClick={[Function]}
              >
                <i
                  className="icon fa fa-pencil edit-icon"
                >
                  <span />
                </i>
              </div>
            </Icon>
            <Icon
              dataTarget="#modal"
              dataToggle="modal"
              handleClick={[Function]}
              iconClass="fa fa-trash-o delete-icon"
              pointer={true}
            >
              <div
                className="icon-wrapper pointer"
                onClick={[Function]}
              >
                <i
                  className="icon fa fa-trash-o delete-icon"
                  data-target="#modal"
                  data-toggle="modal"
                >
                  <span />
                </i>
              </div>
            </Icon>
          </div>
        </div>
      </ListItem>
    </div>
  </List>
</Recipes>
`;

exports[`Recipes list component  render as expected when created recipes nav link is clicked  2`] = `
<Recipes
  handleClick={[Function]}
  onDeleteIconClicked={[Function]}
  onEditIconCliked={[Function]}
  onFavIconClicked={[Function]}
  recipes={
    Array [
      Object {},
    ]
  }
  type="createdRecipes"
>
  <List
    className="recipe-list"
    style={Object {}}
  >
    <div
      className="col-12 recipe-list"
      style={Object {}}
    >
      <h4
        className="display-4 bold header"
      >
        my recipes
      </h4>
      <ListItem
        className="recipe-list-item"
        handleClick={[Function]}
        key="0"
      >
        <div
          className="lead list-item recipe-list-item"
        >
          <div>
            <Icon
              handleClick={[Function]}
              iconClass="fa fa-pencil edit-icon"
              pointer={true}
            >
              <div
                className="icon-wrapper pointer"
                onClick={[Function]}
              >
                <i
                  className="icon fa fa-pencil edit-icon"
                >
                  <span />
                </i>
              </div>
            </Icon>
            <Icon
              dataTarget="#modal"
              dataToggle="modal"
              handleClick={[Function]}
              iconClass="fa fa-trash-o delete-icon"
              pointer={true}
            >
              <div
                className="icon-wrapper pointer"
                onClick={[Function]}
              >
                <i
                  className="icon fa fa-trash-o delete-icon"
                  data-target="#modal"
                  data-toggle="modal"
                >
                  <span />
                </i>
              </div>
            </Icon>
          </div>
        </div>
      </ListItem>
    </div>
  </List>
</Recipes>
`;

exports[`Recipes list component  render as expected when delete and edit icon are clicked  1`] = `
<Recipes
  handleClick={[Function]}
  onDeleteIconClicked={[Function]}
  onEditIconCliked={[Function]}
  onFavIconClicked={[Function]}
  recipes={
    Array [
      Object {},
    ]
  }
  type="createdRecipes"
>
  <List
    className="recipe-list"
    style={Object {}}
  >
    <div
      className="col-12 recipe-list"
      style={Object {}}
    >
      <h4
        className="display-4 bold header"
      >
        my recipes
      </h4>
      <ListItem
        className="recipe-list-item"
        handleClick={[Function]}
        key="0"
      >
        <div
          className="lead list-item recipe-list-item"
        >
          <div>
            <Icon
              handleClick={[Function]}
              iconClass="fa fa-pencil edit-icon"
              pointer={true}
            >
              <div
                className="icon-wrapper pointer"
                onClick={[Function]}
              >
                <i
                  className="icon fa fa-pencil edit-icon"
                >
                  <span />
                </i>
              </div>
            </Icon>
            <Icon
              dataTarget="#modal"
              dataToggle="modal"
              handleClick={[Function]}
              iconClass="fa fa-trash-o delete-icon"
              pointer={true}
            >
              <div
                className="icon-wrapper pointer"
                onClick={[Function]}
              >
                <i
                  className="icon fa fa-trash-o delete-icon"
                  data-target="#modal"
                  data-toggle="modal"
                >
                  <span />
                </i>
              </div>
            </Icon>
          </div>
        </div>
      </ListItem>
    </div>
  </List>
</Recipes>
`;

exports[`Recipes list component  render as expected when delete and edit icon are clicked  2`] = `
<Recipes
  handleClick={[Function]}
  onDeleteIconClicked={[Function]}
  onEditIconCliked={[Function]}
  onFavIconClicked={[Function]}
  recipes={
    Array [
      Object {},
    ]
  }
  type="createdRecipes"
>
  <List
    className="recipe-list"
    style={Object {}}
  >
    <div
      className="col-12 recipe-list"
      style={Object {}}
    >
      <h4
        className="display-4 bold header"
      >
        my recipes
      </h4>
      <ListItem
        className="recipe-list-item"
        handleClick={[Function]}
        key="0"
      >
        <div
          className="lead list-item recipe-list-item"
        >
          <div>
            <Icon
              handleClick={[Function]}
              iconClass="fa fa-pencil edit-icon"
              pointer={true}
            >
              <div
                className="icon-wrapper pointer"
                onClick={[Function]}
              >
                <i
                  className="icon fa fa-pencil edit-icon"
                >
                  <span />
                </i>
              </div>
            </Icon>
            <Icon
              dataTarget="#modal"
              dataToggle="modal"
              handleClick={[Function]}
              iconClass="fa fa-trash-o delete-icon"
              pointer={true}
            >
              <div
                className="icon-wrapper pointer"
                onClick={[Function]}
              >
                <i
                  className="icon fa fa-trash-o delete-icon"
                  data-target="#modal"
                  data-toggle="modal"
                >
                  <span />
                </i>
              </div>
            </Icon>
          </div>
        </div>
      </ListItem>
    </div>
  </List>
</Recipes>
`;

exports[`Recipes list component  render as expected when type equals favRecipes  1`] = `
<Recipes
  handleClick={[Function]}
  onDeleteIconClicked={[Function]}
  onEditIconCliked={[Function]}
  onFavIconClicked={[Function]}
  recipes={
    Array [
      Object {},
    ]
  }
  type="favRecipes"
>
  <List
    className="recipe-list"
    style={Object {}}
  >
    <div
      className="col-12 recipe-list"
      style={Object {}}
    >
      <h4
        className="display-4 bold header"
      >
        my favorite recipes
      </h4>
      <ListItem
        className="recipe-list-item"
        handleClick={[Function]}
        key="0"
      >
        <div
          className="lead list-item recipe-list-item"
        >
          <div>
            <Icon
              handleClick={[Function]}
              iconClass="fa fa-heart fav-icon"
              pointer={true}
            >
              <div
                className="icon-wrapper pointer"
                onClick={[Function]}
              >
                <i
                  className="icon fa fa-heart fav-icon"
                >
                  <span />
                </i>
              </div>
            </Icon>
          </div>
        </div>
      </ListItem>
    </div>
  </List>
</Recipes>
`;

exports[`Recipes list component  render as expected when type equals favRecipes  2`] = `
<Recipes
  handleClick={[Function]}
  onDeleteIconClicked={[Function]}
  onEditIconCliked={[Function]}
  onFavIconClicked={[Function]}
  recipes={
    Array [
      Object {},
    ]
  }
  type="favRecipes"
>
  <List
    className="recipe-list"
    style={Object {}}
  >
    <div
      className="col-12 recipe-list"
      style={Object {}}
    >
      <h4
        className="display-4 bold header"
      >
        my favorite recipes
      </h4>
      <ListItem
        className="recipe-list-item"
        handleClick={[Function]}
        key="0"
      >
        <div
          className="lead list-item recipe-list-item"
        >
          <div>
            <Icon
              handleClick={[Function]}
              iconClass="fa fa-heart fav-icon"
              pointer={true}
            >
              <div
                className="icon-wrapper pointer"
                onClick={[Function]}
              >
                <i
                  className="icon fa fa-heart fav-icon"
                >
                  <span />
                </i>
              </div>
            </Icon>
          </div>
        </div>
      </ListItem>
    </div>
  </List>
</Recipes>
`;

exports[`Recipes list component  to match empty snapshot  1`] = `
<List
  className="recipe-list"
  style={Object {}}
>
  <div
    className="display-3 text-uppercase msg-display"
  >
    <h1>
      Sorry..
      <br />
      No recipe was found.
    </h1>
  </div>
</List>
`;
